src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -33,6 +33,74 @@ constexpr const auto TsfRedrawInterval = std::chrono::milliseconds(100);
 // The minimum delay between updating the locations of regex patterns
 constexpr const auto UpdatePatternLocationsInterval = std::chrono::milliseconds(500);
 
+static constexpr std::optional<Terminal::SelectionDirection> ConvertVKeyToSelectionDirection(WORD vkey)
+{
+    switch (vkey)
+    {
+    case VK_LEFT:
+    case VK_HOME:
+        return Terminal::SelectionDirection::Left;
+    case VK_RIGHT:
+    case VK_END:
+        return Terminal::SelectionDirection::Right;
+    case VK_UP:
+    case VK_PRIOR:
+        return Terminal::SelectionDirection::Up;
+    case VK_DOWN:
+    case VK_NEXT:
+        return Terminal::SelectionDirection::Down;
+    default:
+        return std::nullopt;
+    }
+}

```

Comment:
This abstraction is only called by one other function. If you inline it like so it'll be immediately obvious which value will be returned for any given input and it saves you a third of the code:
```cpp
using UpdateSelectionParams = std::optional<std::pair<Terminal::SelectionDirection, Terminal::SelectionExpansion>>;

static constexpr UpdateSelectionParams ConvertKeyEventToUpdateSelectionParams(const ControlKeyStates mods, const WORD vkey)
{
    if (mods.IsShiftPressed() && !mods.IsAltPressed())
    {
        if (mods.IsCtrlPressed())
        {
            switch (vkey)
            {
            case VK_LEFT:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Left, Terminal::SelectionExpansion::Word };
            case VK_RIGHT:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Right, Terminal::SelectionExpansion::Word };
            case VK_HOME:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Left, Terminal::SelectionExpansion::Buffer };
            case VK_END:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Right, Terminal::SelectionExpansion::Buffer };
            default:
                [[fallthrough]];
            }
        }
        else
        {
            switch (vkey)
            {
            case VK_HOME:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Left, Terminal::SelectionExpansion::Viewport };
            case VK_END:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Right, Terminal::SelectionExpansion::Viewport };
            case VK_PRIOR:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Up, Terminal::SelectionExpansion::Viewport };
            case VK_NEXT:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Down, Terminal::SelectionExpansion::Viewport };
            case VK_LEFT:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Left, Terminal::SelectionExpansion::Char };
            case VK_RIGHT:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Right, Terminal::SelectionExpansion::Char };
            case VK_UP:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Up, Terminal::SelectionExpansion::Char };
            case VK_DOWN:
                return UpdateSelectionParams{ std::in_place, Terminal::SelectionDirection::Down, Terminal::SelectionExpansion::Char };
            default:
                [[fallthrough]];
            }
        }
    }
    return std::nullopt;
}
```

* `[[fallthrough]]`
* `std::pair` if you got exactly 2 items in a tuple (the most common case anyways)
* idk if you like the above more, but I prefer it (it's easier to understand IMO)
* I'd suggest making this a private member function to save yourself from writing `Terminal::` again and again.